contracts/libraries/FullMath.sol

```diff
@@ -38,10 +38,14 @@ library FullMath {
         uint256 d
     ) internal pure returns (uint256) {
         (uint256 l, uint256 h) = fullMul(x, y);
+
         uint256 mm = mulmod(x, y, d);
         if (mm > l) h -= 1;
         l -= mm;
-        require(h < d, 'FullMath::mulDiv: overflow');
+
+        if (h == 0) return l / d;
```

Comment:
because of the issue opened the other day, i found some fairly unintuitive results from fullMul where h was nonzero for cases where i expected the high bits of a 512 bit number to be 0
https://github.com/Uniswap/uniswap-lib/issues/24#issuecomment-727264834

Reply:
errrr, ignore this
i miscomputed
```
        uint256 mm = mulmod(x, y, uint256(-1)); // 0
```

lol

Reply:
maybe still true, h seems to be uint256(-1)

Reply:
the echidna test i added gives me some confidence: https://github.com/Uniswap/uniswap-lib/pull/30/files#diff-4b62046edb555008f8d357d22ff56a1d1371b164663bcac17a0f4b62418d93d0R1